scripts/calculateRevenue/calculateRevenue.ts

```diff
@@ -0,0 +1,79 @@
+import calculateRevenueHandlers from './protocols'
+import { readFileSync, writeFileSync } from 'fs'
+import yargs from 'yargs'
+import { protocols, Protocol, RevenueResult } from '../types'
+
+async function main(args: ReturnType<typeof parseArgs>) {
+  const eligibleAddresses = readFileSync(args['input-addresses'], 'utf-8')
+    .split('\n')
+    .filter((address) => address !== '')
+
+  const handler = calculateRevenueHandlers[args['protocol-id'] as Protocol]
+
+  const allResults: Record<string, RevenueResult> = {}
+
+  for (let i = 0; i < eligibleAddresses.length; i++) {
+    const address = eligibleAddresses[i]
+    console.log(
+      `Calculating revenue for ${address} (${i + 1}/${eligibleAddresses.length})`,
+    )
+    const userResult = await handler(
```

Comment:
is this await better to do it serially? or can we use Promise.all?

Reply:
I worry about running into 429s if we parallelize all these calls. Eventually, the helper methods responsible for fetching data will be memoized/cached, so subsequent calls should be much faster.